scripts/vae_pancancer.py

```diff
@@ -0,0 +1,222 @@
+args = parser.parse_args()
+
+# Set hyper parameters
+learning_rate = float(args.learning_rate)
+batch_size = int(args.batch_size)
+epochs = int(args.epochs)
+kappa = float(args.kappa)
+depth = int(args.depth)
+first_layer = int(args.first_layer)
+output_filename = args.output_filename
+
+original_dim = 5000
+latent_dim = 100
+epsilon_std = 1.0
+beta = K.variable(0)
+
+if depth == 2:
+    latent_dim2 = int(first_layer)
+
+# Random seed
+seed = int(np.random.randint(low=0, high=10000, size=1))
+np.random.seed(seed)
+
+
+# Function for reparameterization trick to make model differentiable
+def sampling(args):
+
+    # Function with args required for Keras Lambda function
+    z_mean, z_log_var = args
+
+    # Draw epsilon of the same shape from a standard normal distribution
+    epsilon = K.random_normal(shape=tf.shape(z_mean), mean=0.,
+                              stddev=epsilon_std)
+
+    # The latent vector is non-deterministic and differentiable
+    # in respect to z_mean and z_log_var
+    z = z_mean + K.exp(z_log_var / 2) * epsilon
+    return z
+
+
+class CustomVariationalLayer(Layer):
+    """
+    Define a custom layer that learns and performs the training
+
+    """
+    def __init__(self, **kwargs):
+        # https://keras.io/layers/writing-your-own-keras-layers/
+        self.is_placeholder = True
+        super(CustomVariationalLayer, self).__init__(**kwargs)
+
+    def vae_loss(self, x_input, x_decoded):
+        reconstruction_loss = original_dim * \
+                              metrics.binary_crossentropy(x_input, x_decoded)
+        kl_loss = - 0.5 * K.sum(1 + z_log_var_encoded -
+                                K.square(z_mean_encoded) -
+                                K.exp(z_log_var_encoded), axis=-1)
+        return K.mean(reconstruction_loss + (K.get_value(beta) * kl_loss))
+
+    def call(self, inputs):
+        x = inputs[0]
+        x_decoded = inputs[1]
+        loss = self.vae_loss(x, x_decoded)
+        self.add_loss(loss, inputs=inputs)
+        # We won't actually use the output.
+        return x
+
+
+class WarmUpCallback(Callback):
+    def __init__(self, beta, kappa):
+        self.beta = beta
+        self.kappa = kappa
+
+    # Behavior on each epoch
+    def on_epoch_end(self, epoch, logs={}):
+        if K.get_value(self.beta) <= 1:
+            K.set_value(self.beta, K.get_value(self.beta) + self.kappa)
+
+rnaseq_file = os.path.join('data', 'pancan_scaled_zeroone_rnaseq.tsv')
+rnaseq_df = pd.read_table(rnaseq_file, index_col=0)
+
+# Split 10% test set randomly
+test_set_percent = 0.1
+rnaseq_test_df = rnaseq_df.sample(frac=test_set_percent)
+rnaseq_train_df = rnaseq_df.drop(rnaseq_test_df.index)
+
+# Input place holder for RNAseq data with specific input size
+rnaseq_input = Input(shape=(original_dim, ))
+
+
+# ~~~~~~~~~~~~~~~~~~~~~~
+# ENCODER
+# ~~~~~~~~~~~~~~~~~~~~~~
+# Input layer is compressed into a mean and log variance vector of size
```

Comment:
Does this need to be updated to take into account the two hidden layers? Hard to tell